src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2310,10 +2314,11 @@ private static StringBuilder BuildCommandLine(string executableFileName, string
                 builder.Append('"');
             }
 
-            if (!string.IsNullOrEmpty(arguments))
+            foreach (var arg in arguments)
             {
-                builder.Append(' ');
-                builder.Append(arguments);
+                builder.Append(" \"");
+                builder.Append(arg.Replace("\"","\\\""));
+                builder.Append('"');
```

Comment:
Why is any such escaping still needed, vs just letting dotnet handle `StartInfo.ArgumentList`?

Reply:
It is handled by ArgumentList for Linux, macOS, but Windows uses the Win32 API CreateProcessWithLogon().  Not sure of the history behind this, but preserving the current behavior.